tests/hf/test_dipole.py

```diff
@@ -0,0 +1,361 @@
+            np.array(
+                [
+                    [
+                        [1.42895581, 0.23469918],
+                        [0.23469918, 0.48381955],
+                    ],
+                    [
+                        [0.3203965, 0.47233426],
+                        [0.47233426, 0.79021614],
+                    ],
+                    [[0.0, 0.0], [0.0, 0.0]],
+                ]
+            ),
+        ),
+    ],
+)
+def test_dipole_integrals(symbols, geometry, core, charge, active, core_ref, int_ref):
+    r"""Test that generate_electron_integrals returns the correct values."""
+    mol = Molecule(symbols, geometry, charge=charge)
+    args = [p for p in [geometry] if p.requires_grad]
+    constants, integrals = dipole_integrals(mol, core=core, active=active)(*args)
+
+    for i in range(3):  # loop on x, y, z components
+        assert np.allclose(constants[i], core_ref[i])
+        assert np.allclose(integrals[i], int_ref[i])
+
+
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "charge", "core", "active", "f_ref"),
+    [
+        (
+            ["H", "H", "H"],
+            np.array(
+                [[0.028, 0.054, 0.0], [0.986, 1.610, 0.0], [1.855, 0.002, 0.0]], requires_grad=False
+            ),
+            1,
+            None,
+            None,
+            # x component of fermionic dipole computed with PL-QChem dipole (format is modified)
+            (
+                np.array(
+                    [
+                        -2.869,
+                        0.956224634652776,
+                        0.782727697897828,
+                        -0.532222940905614,
+                        0.956224634652776,
+                        0.782727697897828,
+                        -0.532222940905614,
+                        0.782727697897828,
+                        1.42895581236226,
+                        0.234699175620383,
+                        0.782727697897828,
+                        1.42895581236226,
+                        0.234699175620383,
+                        -0.532222940905614,
+                        0.234699175620383,
+                        0.483819552892797,
+                        -0.532222940905614,
+                        0.234699175620383,
+                        0.483819552892797,
+                    ]
+                ),
+                [
+                    [],
+                    [0, 0],
+                    [0, 2],
+                    [0, 4],
+                    [1, 1],
+                    [1, 3],
+                    [1, 5],
+                    [2, 0],
+                    [2, 2],
+                    [2, 4],
+                    [3, 1],
+                    [3, 3],
+                    [3, 5],
+                    [4, 0],
+                    [4, 2],
+                    [4, 4],
+                    [5, 1],
+                    [5, 3],
+                    [5, 5],
+                ],
+            ),
+        ),
+    ],
+)
```

Comment:
I suggest including a test for a given active space which is 99% the case in actual applications.

Reply:
Also weird that codecov is not complaining.

Reply:
Added test for active space. codecov did not complain because the active space was tested in test_dipole_integrals. Other functions just pass the core and active orbitals to that function.